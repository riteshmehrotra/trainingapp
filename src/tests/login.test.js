import React from 'react';
import { render, fireEvent, getByPlaceholderText, getByText } from '@testing-library/react';
import Login from '../login';
import renderer from 'react-test-renderer';


describe("Login details capture: ", () => {

    let mockCallBack;
    let component;

    beforeEach(() => {
        mockCallBack = jest.fn();
        let { container } = render(<Login onSubmit={mockCallBack} ></Login>);
        component = container;
    })

    afterEach(() => {
        mockCallBack = null;
        component = null;
    })

    test("Test that login is successfully submitted when name is provided", () => {
        let input = getByPlaceholderText(component, 'Enter your name');
        fireEvent.change(input, { target: { value: 'Ritesh' } });
        expect(input.value).toBe("Ritesh");
        let button = getByText(component, 'Enter').closest('button');
        expect(button).not.toBeDisabled();
        fireEvent.submit(button);
        expect(mockCallBack).toHaveBeenCalledWith("Ritesh");
    });


    test("Test that user cannot submit form when name is not prov ided", () => {
        expect(getByText(component, 'Enter').closest('button')).toBeDisabled();
        let button = getByText(component, 'Enter').closest('button');
        fireEvent.submit(button);
        expect(mockCallBack).not.toHaveBeenCalled();
    })


    test("Test that user cannot submit form when name is provided with empty string", () => {
        let input = getByPlaceholderText(component, 'Enter your name');
        fireEvent.change(input, { target: { value: '                        ' } });
        let button = getByText(component, 'Enter').closest('button')
        expect(button).toBeDisabled();
        fireEvent.submit(button);
        expect(mockCallBack).not.toHaveBeenCalled();
    })


    test("Test snapshot of login form",()=>{
        const tree = renderer.create(<Login onSubmit={mockCallBack}></Login>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})