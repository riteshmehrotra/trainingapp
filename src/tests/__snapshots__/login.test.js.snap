// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login details capture:  Test snapshot of login form 1`] = `
<div
  className="row"
  style={
    Object {
      "justifyContent": "center",
      "marginTop": "15%",
    }
  }
>
  <div
    className="col-lg-6"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        Welcome to TDD with React Training
      </div>
      <div
        className="card-body"
      >
        <form
          className=""
          onSubmit={[Function]}
        >
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="Enter your name"
            type="text"
          />
          <br />
          <button
            className="btn btn-success btn-block"
            disabled={true}
            onClick={[Function]}
            type="submit"
          >
            Enter
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
